jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -1240,4 +1258,54 @@ public String toString()
                 '}';
         }
     }
+
+    public class ContextProcessor implements Request.Processor
+    {
+        private final Request.Processor _processor;
+
+        public ContextProcessor(Request.Processor processor)
+        {
+            _processor = processor;
+        }
+
+        @Override
+        public void process(Request request, Response response, Callback callback) throws Exception
+        {
+            Request contextRequest = wrap(request); // TODO reuse any previous wrapper
```

Comment:
I think this method can be simplified to:
```java
Request contextRequest = wrap(request); // TODO reuse any previous wrapper
Response contextResponse = new ContextResponse(_context, contextRequest, response);
_context.run(() ->
{
    try
    {
        _processor.process(contextRequest, contextResponse, callback);
    }
    catch (Throwable th)
    {
        Response.writeError(contextRequest, contextResponse, callback, th);
    }
}, contextRequest);
```

Reply:
yeah... but I want to avoid creating lambdas on the hot path.